core/src/toga/window.py

```diff
@@ -393,6 +398,38 @@ def size(self, size: SizeT) -> None:
         if self.content:
             self.content.refresh()
 
+    def dirty(self, widget: Widget) -> None:
+        """Mark widget as being 'dirty', so that it gets refreshed
+        when the window lock is lifted
+        """
+        self._dirty.add(widget)
+
+    @property
+    def locked(self) -> bool:
+        """Is the window locked from refreshes?"""
+        return self._locked
+
+    @locked.setter
+    def locked(self, locked: bool) -> None:
```

Comment:
This flag definitely shouldn't be writable. A user shouldn't be able to inadvertently change the locking state. A *getter* makes sense; but I can't see a use case for releasing the lock being a public API (outside of the context manager)